index.rst

```diff
@@ -2,24 +2,180 @@
+
+.. _salobj: ts-salobj.lsst.io
+.. _ts_externalscripts: https://github.com/lsst-ts/ts_externalscripts
+.. _ts_standardscripts: https://github.com/lsst-ts/ts_standardscripts
+
+The usage of SAL Scripts for integration testing comes, mainly, in two different varieties; regular operations that can be run in special conditions for testing purposes and custom made to test a particular condition.
+In the first case, we simply use a regular operation (e.g. slew and take data) with a particular configuration for which we can easily predict the outcome, whereas scripts designed to execute an integration tests have pre-defined predictable behavior.
+The first line of testing consist in making sure the script executes successfully, followed by a verification that the data produced by the different components matches our expectations (see :ref:`verification-artifacts` furthermore).
+
+The most obvious advantage of using regular operations for integration testing is that we are developing and using SAL Script that will be regularly used.
+Most of these operations exercise the vast majority of the system requirements, which makes them a perfect tool to verify them.
+Since the majority of these uses TSSW high-level libraries (e.g. ts_observatory_control), they are extremely maintainable, requiring minimal to no updates when there are underlying changes to the CSCs behavior.
+
+Jupyter Notebooks
+-----------------
+
+Jupyter notebooks provides a complimentary interactive interface into the Rubin Observatory Control system.
+They are used in parallel with SAL Scripts to execute tasks that require some user interaction or a finer degree of control over the operations.
+
+Although they can be very useful in the early development stages of the process, we do not recommend storing any long-term use software in notebooks.
+The main reason is that it is currently note possible to unit test Jupyter notebooks, which can leave to bit rot in short timescales, especially with the rapidly evolving software development one would expect in early commissioning stages.
+
+Exceptional Cases
+-----------------
+
+.. Since part of Telescope and Site Software scope is to deliver a CI/CD system for the observatory control software, this system
+
+Although a substantial number of requirements can be verified with the use of our CI system, we acknowledge that some of them will inevitably demand a more individualized approach and, sometimes, even a more involved process demanding additional data analysis and/or a more descriptive verification methodology.
+
+Ideally, these cases would be addressed by either developing a customized SAL Script to drive the system through the required test steps or, in the more extreme conditions that might require interactive steps, a Jupyter notebook.
+
+Furthermore, any detailed data analysis/methodology should be included in a technical note to be incorporated into the requirement verification artifacts.
+
```

Comment:
So what's your plan to ensure that these specialized hand-run one-off things don't break?

Reply:
Are you referring to the test execution or the analysis? If the test execution is done via Script (e.g. acquiring points model data) then they will be under the umbrella of CI. Notebooks unfortunately are on their own and can only see as representative of a snapshot of the system. For the analysis I guess they refer to a particular test execution and hopefully will have enough information to allow people to reproduce it later for other test executions if they need to.

Reply:
I include a couple phrases clarifying both points (test execution and analysis)..